xdsl/irdl.py

```diff
@@ -470,6 +470,29 @@ def from_pyrdl(pyrdl_def: type[_OpT]) -> OpDef:
                 "adding a 'name' field.")
 
         op_def = OpDef(clsdict["name"])
+        for field_name, field_type in get_type_hints(
+                pyrdl_def, include_extras=True).items():
+            origin = get_origin(field_type)
+            if origin != Annotated:
+                continue
+            args = get_args(field_type)
+
+            if isinstance(args[-1], OperandDef):
+                op_def.operands.append((field_name, args[-1]))
+            elif isinstance(args[-1], VarOperandDef):
```

Comment:
`VarOperandDef` is a subclass of `OperandDef` (same for the following), so you can remove these cases.

Reply:
Done :)